NATS Subscription Controller (#10251)
kyma-project/kyma: components/eventing-controller/README.md

## Overview

This component contains controllers for various CustomResourceDefinitions related to eventing in Kyma. The following controllers come with this container:

> - [controller](https://github.com/kyma-project/kyma/blob/master/components/eventing-controller/cmd/eventing-controller/main.go) which lays down the eventing infrastructure in BEB

Reviewer: The component is named Eventing Controller and includes two controllers: `controller` and `nats-controller`. It that is the naming convention we want to take, let's stick to it throughout the doc as later in the docs, I can see `controller` is referred to as Eventing Controller which is quite misleading. Also, let's use either `nats-controller` or NATS (NATS, not Nats, right?) Controller:
![image](https://user-images.githubusercontent.com/40655785/104023731-5b2b4600-51c2-11eb-9428-0a1f07d88c3a.png)

> - [controller](https://github.com/kyma-project/kyma/blob/master/components/eventing-controller/cmd/eventing-controller/main.go) which lays down the eventing infrastructure in BEB

Author: @kazydek You're definitely right. IMHO, the best approach is to change the code/directories of `controller` to be `beb-controller` and change the documentation then based on that.
Is it fine if we resolve this for now and create a separate ticket for renaming the code occurrences as well as the docs, as it will be a technical task with intensive technical testing for this breaking change?
CC: @sayanh

> - [controller](https://github.com/kyma-project/kyma/blob/master/components/eventing-controller/cmd/eventing-controller/main.go) which lays down the eventing infrastructure in BEB

Reviewer: It makes sense to change the names to be more descriptive in the future. For now, however, if we have the distinction between `controller` and `nats-controller`, I'd suggest sticking to it throughout the doc. I already added comments that change it :)

    make deploy-local-dry-run
    ```

- For `nats-controller`, run:

    ```sh
    make deploy-eventing-controller-nats-local

    ## To verify all the manifests processed by Kustomize, without applying them to the cluster, use the make target called "deploy-eventing-controller-nats-local-dry-run".    
    make deploy-eventing-controller-nats-local-dry-run
    ```

## Usage 

This section explains how to use the Eventing Controller.

> - The Eventing Controller comes with the following command line argument flags:

Reviewer: ```suggestion
- `controller` comes with the following command line argument flags:
```

    | Flag                    | Description                                           | Default Value   |
    | ----------------------- | ----------------------------------------------------- | --------------- |
    | `metrics-addr`            | The address the metric endpoint binds to.           | `:8080`         |
    | `reconcile-period`        | The period between triggering of reconciling calls. | `10 minutes`    |
    | `enable-debug-logs`       | Enable debug logs.                                  | `false`         |

> - The Nats Controller comes with the following command line argument flags:

Reviewer: ```suggestion
- `nats-controller` comes with the following command line argument flags:
```

    | Flag                    | Description                                           | Default Value |
    | ----------------------- | ----------------------------------------------------- | ------------- |
    | `metrics-addr`            | The address the metric endpoint binds to.           | `:8080`       |
    | `reconcile-period`        | The period between triggering reconciling calls.    | `10 minutes`  |
    | `enable-debug-logs`       | The parameter that enables debug logs.              | `false`       |
    | `max-reconnects`          | The maximum number of reconnection attempts.        | `10 `         |

>     | `reconnect-wait `         | Wait time between reconnection attempts.            | `1 second`    |

Reviewer: ```suggestion
    | `reconnect-wait`         | Wait time between reconnection attempts.            | `1 second`    |
```

> - The Eventing Controller expects the following environment variables:

Reviewer: ```suggestion
- `controller` expects the following environment variables:
```

    | Environment variable       | Description                                                                     |
    | -------------------------- | ------------------------------------------------------------------------------- |
    | **CLIENT_ID**              | The Client ID used to acquire Access Tokens from the Authentication server.     |
    | **CLIENT_SECRET**          | The Client Secret used to acquire Access Tokens from the Authentication server. |
    | **TOKEN_ENDPOINT**         | The Authentication Server Endpoint to provide Access Tokens.                    |
    | **WEBHOOK_CLIENT_ID**      | The Client ID used by webhooks to acquire Access Tokens from Kyma.              |
    | **WEBHOOK_CLIENT_SECRET**  | The Client Secret used by webhooks to acquire Access Tokens from Kyma.          |
    | **WEBHOOK_TOKEN_ENDPOINT** | The Kyma public endpoint to provide Access Tokens.                              |
    | **DOMAIN**                 | The Kyma cluster public domain.                                                 |

> - The NATS Controller expects the following environment variables:

Reviewer: ```suggestion
- `nats-controller` expects the following environment variables:
```

    | reconcile-period        | The period between triggering of reconciling calls. | 10 minutes    |
    | enable-debug-logs       | Enable debug logs.                                  | false         |

- The Nats Controller comes with the following command line argument flags:

    | Flag                    | Description                                         | Default Value |
    | ----------------------- | --------------------------------------------------- | ------------- |

>     | metrics-addr            | The address the metric endpoint binds to.           | :8080         |

Reviewer: ```suggestion
    | `metrics-addr`            | The address the metric endpoint binds to.           | `:8080`         |
```

Reviewer: As per guidelines, we use `code font` for flags (values in tables as well) and **bold** envs.  Could you pls adjust all of them in tables?

>     | reconcile-period        | The period between triggering of reconciling calls. | 10 minutes    |

Reviewer: ```suggestion
    | `reconcile-period`        | The period between triggering reconciling calls. | 10 minutes    |
```

>     | enable-debug-logs       | Enable debug logs.                                  | false         |

Reviewer: ```suggestion
    | enable-debug-logs       | The parameter that enables debug logs.                                  | `false`         |
```

Reviewer: I would start with a noun to adjust it to other descriptions.

>     | max-reconnects          | Maximum number of reconnect attempts.               | 10            |

Reviewer: ```suggestion
    | `max-reconnects`          | The maximum number of reconnection attempts.               | `10 `           |
```

Reviewer: I added `the` just to be consistent with other descriptions. If you prefer to remove it, let's do it at the beginning of all descriptions.

>     | reconnect-wait          | Wait time between reconnect attempts.               | 1 second      |

Reviewer: ```suggestion
    | `reconnect-wait `         | Wait time between reconnection attempts.               | 1 second      |
```

    | WEBHOOK_TOKEN_ENDPOINT | The Kyma public endpoint to provide Access Tokens.                              |
    | DOMAIN                 | The Kyma cluster public domain.                                                 |

> - The Nats Controller expects the following environment variables:

Reviewer: ```suggestion
- The NATS Controller expects the following environment variables:
```

    | Environment variable   | Description                                                                     |
    | ---------------------- | ------------------------------------------------------------------------------- |

>     | NATS_URL               | The URL for Nats server.     |

Reviewer: ```suggestion
    | **NATS_URL**               | The URL for the NATS server.     |
```

Reviewer: Can you **bold** envs in the whole table?
